platform/ios/src/MGLMapView.mm

```diff
@@ -4583,10 +4585,11 @@ - (void)calloutViewWillAppear:(UIView <MGLCalloutView> *)calloutView
                                  UIViewAnimationOptionBeginFromCurrentState)
                      animations:^
      {
+         __typeof__(self) strongSelf = weakSelf;
```

Comment:
@1ec5 you appear to be the original author here — does this fit your expectation?

Reply:
Yes. However, if `weakSelf` is already `nil`, then dereferencing it below will crash. Either return early or turn `initialImplicitCalloutViewOffset` into a property whose getter message you can send more safely to `nil`.

Reply:
Went with “return early” in https://github.com/mapbox/mapbox-gl-native/pull/11771/commits/f194ddd3d21ac9996f94a3d4a7bdc995a0404504.